### ISSUE TITLE
Update to Dart SDK 1.23.0-dev.9.0

### ISSUE THREAD

**Issue by pq** [flutter/flutter#8812]
_(We're currently on dev.4.0.)_

TL;DR because of a new dependency on the `kernel` package (from the `front_end`), we need to take this in a few steps and will need to wait for `dev.8.0` (UPDATE: actually, `dev.9.0`).

The longer story:

Attempting the update today, meets us with this error when we try to build `flutter`:

```
Could not import "package:kernel/analyzer/loader.dart" from "package:front_end/src/fasta/analyzer/element_store.dart": 
No mapping for 'kernel' package when resolving 'package:kernel/analyzer/loader.dart'.
```

To get us past this we need to do a few things.

* [x] add the `kernel` package to the Dart SDK sources ([cl/2756593002](https://codereview.chromium.org/2756593002/))

and once that lands and the `kernel` sources are in `dev.9.0`, we want to

* [ ] introduce a dev dependency (like we do already for `analyzer` and `front_end`) in `flutter_tools` and elsewhere

FYI @kasperl as this impacts/delays landing strong-mode related analysis options changes

**Comment by whesse** [comment]
This will be in -dev.9.0 when it lands - I just released -dev.8.0 which was requested for Google3 integration.

**Comment by pq** [comment]
Cool.  Thanks @whesse !
